### ISSUE TITLE
Inference speed is abnormally slow

### ISSUE THREAD

**Issue by samayala22** [open-mmlab/mmocr#335]
(All the files I will be referencing are in this google [drive](https://drive.google.com/drive/folders/1AsFeGLUnk4XCpYoetHJLMqHrxZMCReNJ?usp=sharing))

I wrote a user friendly API interface so that people could try and put into production the algorithms in mmocr much more easily (ocr_main.py). Here's an example:

```python
from ocr_main import MMOCR

ocr = MMOCR(textdet= 'DB_r18', textrecog='CRNN')
results = ocr.readtext('.\\demo\\demo_text_det.jpg',out_img='.\\demo\\testing_out.jpg',details=False)
print(results)
```

Unfortunately, when testing (test.py), the inference speed was very slow. About 5 times slower compared to the PaddlePaddleOCR DB+CRNN implementation. When using DB+NRTR in mmocr,  it took 10 min to recognise the demo_text_det.jpg 50 times !

- mmocr_DB_CRNN.prof : profiling results of my test script
- ucr_DB_CRNN.prof : profiling results of the [UCR](https://github.com/DocYard-ai/UCR) implementation which uses PaddlePaddle's scripts.

As you can see when comparing the profiling results, mmocr is making too many calls to some pytorch built-in methods.

So I'd like to know if something is wrong with my interface, or is it because of the mmocr implementation or maybe something else ?

Once the issue is solved, I will create a PR to submit my ocr_main.py file.

_Note: Put the ocr_main.py and test.py files in the mmocr/mmocr directory (same dir where there is apis, core, datasets,...) and call the test.py script from the root mmocr directory (because mmdet is sensitive to pwd)._

Testing environment:
- Windows 10 (yes I managed to compile lanms)
- i7 9750h and RTX 2060
- Python 3.9
- Pytorch 1.8
- Cuda 11.1

**Comment by gaotongxiao** [comment]
Sorry for the late reply. I have tested your script on my machine (ubuntu 16.04, gtx 1060, i7 8700), and it took ~15s on DB+CRNN, 3.5 min on DB+NRTR. There is performance loss on Windows since MMCV does not officially support windows yet.
As for UCR, I wasn't able to download their pretrained models due to the broken links. 
We have already noticed the inference time problem. Currently, we already have a plan to remove some redundant pipelines; also, we started to provide some deployment options aiming at reducing the inference speed. We will also analyze your .prof and try to reduce unnecessary calls.

**Comment by samayala22** [comment]
Okay thanks for the reply ! I guess the fact that MMCV is not fully supported on Windows could be the reason why it's so slow. I suspect UCR to be that fast because they translated the models that came from PPOCR, so basically with an ultra-lightweight backbone (which I think is a modified version of MobileNetV3). 

If you're interested in playing with UCR a bit, you can find the models in this [Google Drive](https://drive.google.com/drive/folders/10fje5HJygHlrhLU-KuNnAk7Rg3wUalTV?usp=sharing). You simply need to put the `.ucr` folder in your username directory (In windows it's usually under 'C:\Users\xxxxx'). Basically in the directory that is returned by `os.path.expanduser("~")`. And when using UCR, call it like this:

```python
from ucr import UCR
ocr = UCR(lang="en_number", device="cuda",det_algorithm='DB')
result = ocr.predict(img, output='output_img', save_image=True,return_type='list',save_tsv=False)
```

All the arguments for `ocr.predict` are detailed in the `ucr.py` file. The repo is interesting as it has some well optimised python files related to model architecture and post-processing. There's even a python version of lanms.
